contracts/src/libraries/token/erc677.cairo

```diff
@@ -0,0 +1,41 @@
+use starknet::ContractAddress;
+
+#[abi]
+trait IERC677Receiver {
+    fn on_token_transfer(sender: ContractAddress, value: u256, data: Array<felt252>);
+    // implements EIP-165, where function selectors are defined by Ethereum ABI using the ethereum function signatures
+    fn supports_interface(interface_id: u32) -> bool;
+}
+
+
+mod ERC677 {
+    use starknet::ContractAddress;
+    use chainlink::libraries::token::erc20::ERC20;
+    use array::ArrayTrait;
+    use array::SpanTrait;
+    use clone::Clone;
+    use array::ArrayTCloneImpl;
+
+    use super::IERC677ReceiverDispatcher;
+    use super::IERC677ReceiverDispatcherTrait;
+
+    const IERC677_RECEIVER_ID: u32 = 0xa53f2491_u32;
+
+    #[event]
+    fn Transfer(from: ContractAddress, to: ContractAddress, value: u256, data: Array<felt252>) {}
+
+    fn transfer_and_call(to: ContractAddress, value: u256, data: Array<felt252>) -> bool {
+        let sender = starknet::info::get_caller_address();
+
+        ERC20::transfer(to, value);
+        Transfer(sender, to, value, data.clone());
+
+        let receiver = IERC677ReceiverDispatcher { contract_address: to };
+
+        let supports = receiver.supports_interface(IERC677_RECEIVER_ID);
+        // if supports {
+        //     receiver.on_token_transfer(sender, value, data);
+        // }
```

Comment:
Can you re-enable this, finish the tests then mark them as `#[ignore]` for now (and leave a comment that it's blocked on nested call support in cairo-test)

Reply:
(https://cairo-book.github.io/ch08-01-how-to-write-tests.html#ignoring-some-tests-unless-specifically-requested)

Reply:
Sounds good

Reply:
Btw I just found out that they now support contract calls within cairo-test in this PR https://github.com/starkware-libs/cairo/pull/2912/files, but it's part of the new alpha release so we can't include the tests for it unless we update to 0v1.0.0-alpha.7 (which isn't supported on mainnet or starknet-devnet) yet.

Reply:
alpha.7 was released 2 weeks ago, this will be in alpha.8